onnxruntime/test/python/onnxruntime_test_python.py

```diff
@@ -43,6 +43,65 @@ def testSetProviders(self):
             # confirm only CPU Provider is registered now.
             self.assertEqual(['CPUExecutionProvider'], sess.get_providers())
 
+    def testSetProvidersWithOptions(self):
```

Comment:
can you also add some tests for some of error cases?

Reply:
I've added some tests for error cases.

Reply:
I'm also thinking about how to test other two provider options, "cuda_mem_limit" and "arena_extend_strategy". 
One approach I have currently is that we need to add "get" methods at the leve of ExecutionProvier, such as CUDAExecutionProvider class, and make it capable of returning provider options configuration.  And at the session level, we can simply call this "get" method to get the config of ep. 
By doing so, not only users can check the current config of the ep underneath but also we can leverage this to write python unittest.

Reply:
In order to better test setting "cuda_mem_limit" and "arena_extend_strategy", it makes sense to add capability of getting ep's configuration. 
Therefore I've added an approach to get execution provider's configuration or options.
In IExecutionProvider class, it has a private variable to hold EP's configuration. From session level, it can call related methods to get all current EPs' configurations.
It's better to add another Python API to get EPs' configurations, right now I've added it as sess.get_provider_options(), however, if we don't want to expose this Python API, I can still think other solutions to test it without exposing this sess.get_provider_options().